department-of-veterans-affairs/abd-vro
# Mcp 2546 - Filter Lighthouse FHIR resource using _lastUpdated search parameter

Provided additional FHIR resource filtering capability by using _lastUpdated.

## What was the problem?
We are looking for possible additional Lighthouse FHIR data filtering to reduce the unneeded data.

Associated tickets or Slack threads:
MCP-2546

## How does this fix it?[^1]
Add the capability to use _lastUpdated search parameter to filter old data.

## How to test this PR
- Step 1. Build and launch the app. Use Full Health Assessment endpoint. Look into svc-lighthouse-api log, it should show something like
   Get FHIR data from https://sandbox-api.va.gov/services/fhir/v0/r4/MedicationRequest?patient=9000682&_count=100

- Step 2. Add lh.filterLastUpdatedDate=2020-12-21 in application.properties in svc-lighthouse-api main resource directory, and repeat Step 1. In svc-lighthouse-api, it should show something like,
  Get FHIR data from https://sandbox-api.va.gov/services/fhir/v0/r4/Condition?patient=9000682&_lastUpdated=ge2020-12-21&category=encounter-diagnosis&_count=100


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
